contracts/modules/TransferManager/GeneralTransferManager.sol

```diff
@@ -192,16 +172,38 @@ contract GeneralTransferManager is ITransferManager {
     */
     function modifyWhitelist(
         address _investor,
-        uint256 _fromTime,
-        uint256 _toTime,
-        uint256 _expiryTime,
-        bool _canBuyFromSTO
+        uint64 _fromTime,
+        uint64 _toTime,
+        uint64 _expiryTime,
+        uint8 _canBuyFromSTO
     )
         public
         withPerm(WHITELIST)
     {
-        //Passing a _time == 0 into this function, is equivalent to removing the _investor from the whitelist
-        whitelist[_investor] = TimeRestriction(_fromTime, _toTime, _expiryTime, _canBuyFromSTO);
+        _modifyWhitelist(_investor, _fromTime, _toTime, _expiryTime, _canBuyFromSTO);
+    }
+
+    /**
+    * @notice Adds or removes addresses from the whitelist.
+    * @param _investor is the address to whitelist
+    * @param _fromTime is the moment when the sale lockup period ends and the investor can freely sell his tokens
+    * @param _toTime is the moment when the purchase lockup period ends and the investor can freely purchase tokens from others
+    * @param _expiryTime is the moment till investors KYC will be validated. After that investor need to do re-KYC
+    * @param _canBuyFromSTO is used to know whether the investor is restricted investor or not.
+    */
+    function _modifyWhitelist(
+        address _investor,
+        uint64 _fromTime,
+        uint64 _toTime,
+        uint64 _expiryTime,
+        uint8 _canBuyFromSTO
+    )
+        internal
+    {
+        if (whitelist[_investor].added == uint8(0)) {
```

Comment:
missing `require(_investor != address(0))`

Reply:
Will add, although this is related to the question as to whether or not we should be whitelisting the issuance address (in this implementation not necessary, in the previous implementation it was).